crates/rpc/rpc/src/eth/filter.rs

```diff
@@ -1078,6 +1098,75 @@ impl<
 
         Ok(self.next.pop_front())
     }
+
+    /// Process large range headers
+    async fn process_large_range(
+        &mut self,
+        range_headers: Vec<SealedHeader<<Eth::Provider as HeaderProvider>::Header>>,
+    ) -> Result<Option<ReceiptBlockResult<Eth::Provider>>, EthFilterError> {
+        // Split headers into chunks
+        let chunk_size = std::cmp::max(range_headers.len() / DEFAULT_PARALLEL_CONCURRENCY, 1);
+        let mut header_chunks = Vec::new();
+        for chunk_start in (0..range_headers.len()).step_by(chunk_size) {
+            let chunk_end = std::cmp::min(chunk_start + chunk_size, range_headers.len());
+            header_chunks.push(range_headers[chunk_start..chunk_end].to_vec());
+        }
+
+        // Process chunks in parallel
+        let mut tasks = Vec::new();
+        for (chunk_idx, chunk_headers) in header_chunks.into_iter().enumerate() {
+            let filter_inner = self.filter_inner.clone();
+            let task = async move {
+                let mut chunk_results = Vec::new();
+
+                for header in chunk_headers {
+                    // Fetch directly from provider - RangeMode is used for older blocks unlikely to
+                    // be cached
+                    let receipts =
+                        match filter_inner.provider().receipts_by_block(header.hash().into())? {
+                            Some(receipts) => Arc::new(receipts),
+                            None => continue, // No receipts found
+                        };
```

Comment:
so we'd need to spawn_blocking this here in particular or the entire async scope